### PR TITLE
Added type check to webhook signature

### PR DESCRIPTION
**Pull request by tbobkermux**
I've implemented a conditional check to see if the signature header is a string or if it is an array and removed the error to upgrade axios package to the latest version.

### CODE REVIEW

**Review by erikpena — CHANGES_REQUESTED** [review]
Would of helped if I click the "submit review" button 🙄

File: controllers/mux.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { Context } from 'koa';
 
 import { getConfig } from '../services/mux';
 import pluginId from '../admin/src/pluginId';
+import util from "util";
+import { sign } from 'node:crypto';
```
[the remark this answers is not in the document]
Comment (tbobkermux) [reply controllers/mux.ts:8]: Yes I think these got added automatically somewhere. Removing them.

File: controllers/mux.ts [hunk 2]
```diff
@@ -73,16 +75,32 @@ const muxWebhookHandler = async (ctx:Context) => {
...
 
+
   const config = await getConfig('general');
 
-  if(sig === undefined || sig === '') {
+  if(sig === undefined || sig === '' || (Array.isArray(sig) && sig.length < 0)) {
     ctx.throw(401, 'Webhook signature is missing');
   }
+  
+  if(Array.isArray(sig) && sig.length > 0) {
```
[the remark this answers is not in the document]
Comment (tbobkermux) [reply controllers/mux.ts:85]: Exactly, a mistake. Amended.

File: controllers/mux.ts [hunk 3]
```diff
@@ -73,16 +75,32 @@ const muxWebhookHandler = async (ctx:Context) => {
...
+  let sig2;
+
+  if(Array.isArray(sig)){
+    sig2 = sig[0];
+  }
+  else{
+    sig2 = sig;
+  }
+
+  const sig3 = sig2;
```
[the remark this answers is not in the document]
Comment (tbobkermux) [reply controllers/mux.ts]: Yes, I simplified it and rid of the sig2/3.

**Comment by dylanjha** [comment]
This is unrelated to the PR, but I noticed we're doing `JSON.stringify(body)` here to get the raw webhook request body. Is there a way we can access the raw body directly?

```
Webhooks.verifyHeader(JSON.stringify(body), sig, config.webhook_signing_secret);
```

This is potentially problematic because of the following scenario

1. Mux creates the signature based off the raw stringified JSON body
2. The webserver that receives the request parses the string and turns it into a JSON object
3. Application code does JSON.stringify on the JSON object to convert it back into a string

The problem is that the string in (1) and the string in (3) *should* be the same, but they're not guaranteed to be, and if they are not exactly the same, then the signature verification will fail.

Getting the raw request body eliminates the risk of this.

This is a common problem, and we document how to handle this in the node SDK: https://github.com/muxinc/mux-node-sdk#verifying-webhook-signatures and the elixir SDK: https://github.com/muxinc/mux-elixir#usage-in-phoenix

**Comment by erikpena** [comment]
> This is unrelated to the PR, but I noticed we're doing `JSON.stringify(body)` here to get the raw webhook request body. Is there a way we can access the raw body directly?
> 
> ```
> Webhooks.verifyHeader(JSON.stringify(body), sig, config.webhook_signing_secret);
> ```
> 
> This is potentially problematic because of the following scenario
> 
> 1. Mux creates the signature based off the raw stringified JSON body
> 2. The webserver that receives the request parses the string and turns it into a JSON object
> 3. Application code does JSON.stringify on the JSON object to convert it back into a string
> 
> The problem is that the string in (1) and the string in (3) _should_ be the same, but they're not guaranteed to be, and if they are not exactly the same, then the signature verification will fail.
> 
> Getting the raw request body eliminates the risk of this.
> 
> This is a common problem, and we document how to handle this in the node SDK: https://github.com/muxinc/mux-node-sdk#verifying-webhook-signatures and the elixir SDK: https://github.com/muxinc/mux-elixir#usage-in-phoenix

Great question.  What I can say is that Strapi's request apis are based on Koa.js (https://koajs.com/).  It might be possible to capture the raw body and grab the string up front.  That would have to be looked at.  But I agree, it's possible that stringifying a json object might change the order of the properties.

**Comment by erikpena** [comment]
Quick update, so the webhooks bit looks like it is addressed.  However, and with more concern, it seems that something has changed with how Strapi handles plugins and exposing settings in the admin dashboard.  What's happening is that "General" links in the Strapi dashboard will show initially allowing a user to go in and configure Strapi (including the Mux Video Uploader).  However, after saving, the whole "General" section on the left menu of Strapi disappears.  This includes Marketplace, Plugins and all of the settings (not just Mux Video Uploader).

We've reached out to Strapi to see if we can get some understanding as to what is happening here.  Will keep this thread apprised!

**Comment by erikpena** [comment]
With Strapi's help, we were able to resolve the permission issues and it's working 🎉 .

In summary, here is what we've done:
- Added typing to resolve issues with Typescript compiling Koa.js types (https://github.com/muxinc/strapi-plugin-mux-video-uploader/issues/5)
- Updated webhookhandler to handle new type provided by Koa.js (https://github.com/muxinc/strapi-plugin-mux-video-uploader/issues/6)
- Updated peerDependency for Strapi for 3.6.2
- Updated dependencies to use appropriately versioned dependencies

@tbobkermux could you run through some testing before we merge this in?

**Review by erikpena — APPROVED** [review]

**Pull request merged**